backend/Dockerfile

```diff
@@ -25,11 +25,6 @@ RUN if [ "$use_remote_build" = "true" ]; then \
 
 # Compile
 FROM python:3.5 as compiler
-RUN apt-get update -y && \
-    apt-get install --no-install-recommends -y -q default-jdk python3-setuptools python3-dev
-RUN wget https://bootstrap.pypa.io/get-pip.py && python3 get-pip.py
```

Comment:
If we skip the step to upgrade pip, we may end up getting previous flakiness caused by old version of pip cannot retry problem.

Reply:
Thanks for the information. This is important. 
I will restore upgrading pip.
BTW, do you think we should continue using python:3.5 for the sample compilation? Maybe we should switch to 3.7?

Reply:
Yep, 3.7 sounds good to me, I remember I tried to install python 3.5 using miniconda and apparently it is no longer supported.